benchmark/misc/getstringwidth.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const common = require('../common.js');
+
+const bench = common.createBenchmark(main, {
+  type: ['ascii', 'emojiseq', 'fullwidth'],
+  n: [10e4]
+}, {
+  flags: ['--expose-internals']
+});
+
+function main({ n, type }) {
+  const { getStringWidth } = require('internal/readline/utils');
+
+  const str = ({
+    ascii: 'foobar'.repeat(100),
```

Comment:
I don’t see much harm in adding such a benchmark, so I’ve added one with an emoji in the middle. As you point out, the benchmark results still look pretty good.

> That would seem to be a realistic/common-ish case where the current implementation might out-perform this one.

I think it’s still faster because the JS implementation is faster than the C++ one by itself, because it’s quite a bit simpler.